pkg/medusa/reconcile.go

```diff
@@ -179,7 +179,7 @@ func UpdateMedusaMainContainer(dcConfig *cassandra.DatacenterConfig, medusaSpec
 
 // Build the image name and pull policy and add it to a medusa container definition
 func setImage(containerImage *images.Image, container *corev1.Container) {
-	image := containerImage.ApplyDefaults(defaultMedusaImage)
+	image := containerImage.Merge(defaultMedusaImage)
```

Comment:
Hmm, it looks like you have a `defaultMedusaImage` here but you've also done some defaulting in your `Merge` function [here](https://github.com/k8ssandra/k8ssandra-operator/pull/526/files)?

Reply:
The defaults were there already, I did not change the behavior at all.

Reply:
I think we can resolve this by removing the defaulting logic from the Merge function. That means that this piece of logic will be fine to keep.